### ISSUE TITLE
at-loader does not display checking errors

### ISSUE THREAD

**Issue by johnbendi** [s-panferov/awesome-typescript-loader#347]
at-loader just leaves me with the following ambiguous error message:
```
[at-loader] Checking started in a separate process...

[at-loader] Checking finished with 5 errors
```

**Comment by AlexGalays** [comment]
I have the same issue with the latest version and webpack 2 rc3.

It always work the very first time, and sometimes it continues to work for a while mais once it get stuck with the above message, I have to restart the watch.

**Comment by kristaps** [comment]
+1

**Comment by kristaps** [comment]
Versions: awesome-typescript-loader@3.0.0-beta.18, typescript@2.1.5, anything else that could help?

**Comment by tony-gutierrez** [comment]
+1, this is killing me.

**Comment by AlexGalays** [comment]
ts-loader works just fine with webpack 2 by the way :p

**Comment by waynesin** [comment]
+1, setting "silent" in "awesomeTypescriptLoaderOptions" to either ```true``` or ```false``` doesn't seem to have any effect.

**Comment by coffius** [comment]
+1

**Comment by Joe32** [comment]
I can see error detail in browser console.

**Comment by erikstoff** [comment]
+1

**Comment by s-panferov** [comment]
Hello everybody, could you please try the latest version? Or could someone please create a repo? I have output tests and everything seems to work fine.

**Comment by jvanbruegge** [comment]
For me it happens only with the webpack 2 node API, with the cli everything works fine

**Comment by s-panferov** [comment]
If you run webpack by hand, the loader acts in `non-watch` mode and simply pushes all the errors to webpack.

https://github.com/s-panferov/awesome-typescript-loader/blob/master/src/instance.ts#L380-L384

You should expect to see the errors in webpack `stats`.

**Comment by dannypule** [comment]
Getting same issue "[at-loader] Checking finished with 12 errors". Doesn't show a list of errors in the command line. Only shows me what the errors are when I run. In my case, certain dependencies hadn't been installed (angular 2 cookie and redux) because I had done this before updating my angular starter applications package.json from it's original repo. 

In other words: run your app on localhost and check whats in the console. Might give some clues.

**Comment by samal84** [comment]
I had transpileOnly: true in my at-loader options. Changed it back to false, and got the errors back.

**Comment by davidmoshal** [comment]
+1

**Comment by davidmoshal** [comment]
same problem: at-loader: 3.0.8, webpack 2.2.1
updated to: 3.12 / 2.3.2 and problem persists.
ts-loader has same problem.
in both cases errors are reported to browser console, but not terminal.

**Comment by dennisat** [comment]
same problem: at-loader: 3.1.3 webpack 2.4.1
**at least** when I run `tsc -watch` I got the errors

**Comment by ythecombinator** [comment]
Still the same problem here 😢  

Tried setting `transpileOnly` to `false`–as suggested by @samal84–but still no changes.

> Using: **at-loader**: `3.1.3` and **webpack** `2.4.1`

**Comment by ilyes-garifullin** [comment]
I have faced with the same issue and I made a temporary solution:

1. go to node_modules/awesome-typescript-loader/dist/checker
2. open runtime.js file
3. find statement
`if (allDiagnostics.length) {
            console.error(colors.red("\n[" + instanceName + "] Checking finished with " + allDiagnostics.length + " errors"));            
        }`

4. add `allDiagnostics.map(function(err) {
                console.error(colors.red(err.messageText));
            })` right after console.error call inside if (allDiagnostics.length) statement

I will be glad if this help to someone else.

**Comment by Blackbaud-SteveBrush** [comment]
@ilyes-garifullin I wonder if your fix merits a PR? Thanks for the tip!

**Comment by bsoulier** [comment]
Just added the filename, makes it easier to find the error:
`allDiagnostics.map(function (err) {console.error(colors.red(err.file.fileName + ':' + err.messageText)); });`

**Comment by pisrael** [comment]
Same problem here. Not being able to see the errors with webpack --watch.

**Comment by davidm-public** [comment]
+1

**Comment by davidm-public** [comment]
This worked for me:

```
allDiagnostics.map(function (err) {console.error(colors.red(err.file.fileName + ':' + err.messageText)); });
```

    "awesome-typescript-loader": "^3.1.2",
    "typescript": "^2.2.2",
    "webpack": "^2.4.1",

**Comment by pisrael** [comment]
At version 3.2.1 it changes a bit

`"awesome-typescript-loader": "^3.2.1"`

**1.** go to node_modules/awesome-typescript-loader/dist/instance.js

**2.** find statement
```
console.error(colors.red("\n[" + instanceName + "] Checking finished with " + diags.length + " errors"));
```
**3.** Add right below it (inside the same if)
```
diags.map( function (diag) {
     console.error(colors.red(diag.pretty)); 
})
```

**Comment by hdpinto** [comment]
Is anything being done about this? I'm still having this issue and it's now September.

**Comment by arthens** [comment]
I'm trying atl and ran into this problem too. I can always "fix" it by introducing a new error (e.g. removing an import). For some reason this causes the errors to start working again.

Initially I thought it simply didn't show the errors if they were the same or a subset of the previous compilation (which would explain why a introducing a new error fixes it), but after using it for a few days I'm not sure anymore...

**Comment by arthens** [comment]
Ok I did a bit of investigation and realised it was a configuration problem, I'm going to post it in case anyone else did the same mistake.

tl;dr [I was not using `CheckerPlugin`](https://github.com/s-panferov/awesome-typescript-loader#configuration), which caused asyncError to be disabled. Adding the plugin fixed my problem.

Longer explanation:
atl canno requires CheckerPlugin to detech watch mode. If _not_ in watch mode (or in watch mode but no plugin enabled) [it will send the errors to webpack instead of printing them](https://github.com/s-panferov/awesome-typescript-loader/blob/master/src/instance.ts#L446).

 However, webpack will only print errors after bundling assets, but based on the change (and possibly your webpack configuration?) it might decide that no bundling is required, and skip printing errors. A few example of changes that would _not_ trigger webpack:

- changing a typescript method signature
- adding/removing a type
- new lines

 This is because these changes are only relevant to TypeScript. Types are removed when the file is compiled to javascript. The compiled version will look unchanged to webpack... which will probably skip compilation _and_ printing errors.

**Comment by winston01** [comment]
@arthens I added CheckerPlugin and set `watch: true` in webpack.config.js, but I still can't see the error details. Can you add a sample of your config file?

**Comment by arthens** [comment]
@winston01 not sure how useful it's going to be, but [here it is](https://gist.github.com/arthens/76209731af2fec5f9fe048e4c8a686e1)

**Comment by timc13** [comment]
This is still a problem - `CheckerPlugin` or not

**Comment by riq-dfaubion** [comment]
`CheckerPlugin` bandaided the problem for me as well.

**Comment by djyde** [comment]
Same problem!

**Comment by alienriver49** [comment]
I think the pull request, #437, is a bit old compared to the latest in the repository since this is now using [webpack-log](https://github.com/webpack-contrib/webpack-log) to log information. In order to fix this now you'll want to find this in the runtime.js file:
```
function processDiagnostics(_a) {
```

Go to the end of the function and add this line:
```
processedDiagnostics.forEach(function (diag) {
    log.info(diag.pretty);
});
```

I'll look into creating a new pull request for changes similar to this.
